MTA_limit_trends/Comp/run_comp_grad_data_update.py

```diff
@@ -701,27 +711,77 @@ def remove_old_data_from_fits(fits, cut):
 
 if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-m", "--mode", choices = ['flight','test'], required = True, help = "Determine running mode.")
+    parser.add_argument("-d", "--data", help = "Determine Data output file path.")
+    parser.add_argument("--deposit", help = "Determine Deposit input file path.")
+    parser.add_argument("-t",'--dtype', nargs = '*', required = False, choices = ['long', 'short', 'week'], help = "List of data types to generate.")
+    parser.add_argument('--full',help = "Determine whether to full regenerate fits, or use the last recorded time entry.", action=argparse.BooleanOptionalAction)
+    args = parser.parse_args()
+    
+    if args.dtype:
+        DTYPE = args.dtype
+
+    if args.mode == 'test':
+        #Smaller test subset
+        GROUP_MSID_DICT = {
+            'Compgradkodak': ['hrmaavg']
+        }
+
+        if args.full is not None:
+            FULL_GEN = args.full
+        if args.data:
+            DATA_DIR = args.data
+        else:
+            DATA_DIR = f"{os.getcwd()}/test/outTest"
+            os.makedirs(DATA_DIR, exist_ok=True)
+        if args.deposit:
+            DEPOSIT_DIR = args.deposit
+
+        run_comp_grad_data_update()
+
+    elif args.mode == "flight":
 #
 #--- Create a lock file and exit strategy in case of race conditions
 #
-    name = os.path.basename(__file__).split(".")[0]
-    user = getpass.getuser()
-    if os.path.isfile(f"/tmp/{user}/{name}.lock"):
-        sys.exit(f"Lock file exists as /tmp/{user}/{name}.lock. Process already running/errored out. Check calling scripts/cronjob/cronlog.")
-    else:
-        os.system(f"mkdir -p /tmp/{user}; touch /tmp/{user}/{name}.lock")
 
-    run_comp_grad_data_update()
+        name = os.path.basename(__file__).split(".")[0]
+        user = getpass.getuser()
+        if os.path.isfile(f"/tmp/{user}/{name}.lock"):
+            notification = f"Lock file exists as /tmp/{user}/{name}.lock at {time.strftime('%Y:%j:%H:%M:%S',time.localtime())}\n"
+            notification += "Process already running/errored out. Check calling scripts/cronjob/cronlog. Killing old process."
+            print(notification)
+            with open(f"/tmp/{user}/{name}.lock") as f:
+                pid = int(f.readlines()[-1].strip())
+            #Kill old stalling process and remove corresponding lock file.
+            os.remove(f"/tmp/{user}/{name}.lock")
+            os.kill(pid,signal.SIGTERM)
+            #Generate lock file for the current corresponding process
+            os.system(f"mkdir -p /tmp/{user}; echo '{os.getpid()}' > /tmp/{user}/{name}.lock")
+        else:
+            #Previous script run must have completed successfully. Prepare lock file for this script run.
+            os.system(f"mkdir -p /tmp/{user}; echo '{os.getpid()}' > /tmp/{user}/{name}.lock")
+        
+        try:
+            run_comp_grad_data_update()
+        except:
+            traceback.print_exc()
```

Comment:
Of note, the script will now kill the old stalling process if a lock file is discovered.